stagemonitor-zipkin/src/main/java/org/stagemonitor/zipkin/BraveZipkinReportingTracerFactory.java

```diff
@@ -0,0 +1,69 @@
+package org.stagemonitor.zipkin;
+
+import org.stagemonitor.core.CorePlugin;
+import org.stagemonitor.core.StagemonitorPlugin;
+import org.stagemonitor.requestmonitor.RequestMonitorPlugin;
+import org.stagemonitor.requestmonitor.tracing.B3HeaderFormat;
+import org.stagemonitor.requestmonitor.tracing.TracerFactory;
+
+import java.util.concurrent.TimeUnit;
+
+import brave.opentracing.BraveTracer;
+import brave.propagation.Propagation;
+import brave.sampler.Sampler;
+import io.opentracing.Tracer;
+import zipkin.reporter.AsyncReporter;
+import zipkin.reporter.urlconnection.URLConnectionSender;
+
+public class BraveZipkinReportingTracerFactory extends TracerFactory {
+
+	@Override
+	public Tracer getTracer(StagemonitorPlugin.InitArguments initArguments) {
+		final brave.Tracer braveTracer = brave.Tracer.newBuilder()
```

Comment:
this will be mostly the same, except Tracing not tracer. the former allows you to control the default propagation inside brave.